engine/crates/integration-tests/tests/federation/auth/requires_scopes.rs

```diff
@@ -0,0 +1,264 @@
+use super::authenticated::with_secure_schema;
+use integration_tests::openid::{CoreClientExt, OryHydraOpenIDProvider, READ_SCOPE, WRITE_SCOPE};
+
+#[test]
+fn anonymous_does_not_any_scope() {
+    with_secure_schema(|engine| async move {
+        let response = engine.execute("query { mustHaveReadScope }").await;
+        insta::assert_json_snapshot!(response, @r###"
+        {
+          "errors": [
+            {
+              "message": "Invalid OAuth2 scope",
```

Comment:
Nitpick, but a "not authenticated" error here would be closer to the root cause.

Reply:
I wouldn't say closer to root scope. The user has an access token. It's missing the right permissions to access this field. We could replace it with a `Not allowed` message though

Reply:
Ah I didn't get that the user is authenticated. All good then :+1: